test/unittest/statistics/rtps/RTPSStatisticsTests.cpp

```diff
@@ -81,7 +87,11 @@ struct MockListener : IListener
             case DISCOVERED_ENTITY:
                 on_entity_discovery(data.discovery_time());
                 break;
+            case SAMPLE_DATAS:
+                on_sample_datas(data.sample_identity_count());
+                break;
             default:
+                on_unexpected_kind(kind);
```

Comment:
That's a good one. This will generate a warning if gtest catches a callback with unknown discriminant...should we promote it to an error? 🤔

Reply:
Combined with the expectation on the constructor, this makes the test fail if `on_unexpected_kind` is ever called. So it is already an error.

Reply:
Didn't notice that 👍